media/server/gstplayer/source/GstCapabilities.cpp

```diff
@@ -207,28 +207,102 @@ std::vector<std::string> GstCapabilities::getSupportedProperties(MediaSourceType
     for (GList *iter = factories; iter != nullptr && !propertiesToLookFor.empty(); iter = iter->next)
     {
         GstElementFactory *factory = GST_ELEMENT_FACTORY(iter->data);
-        GType elementType = m_gstWrapper->gstElementFactoryGetElementType(factory);
-        if (elementType == G_TYPE_INVALID)
-            continue;
-        gpointer elementClass = m_glibWrapper->gTypeClassRef(elementType);
-        if (elementClass)
+        gpointer elementClass{nullptr};
+        GstPluginFeature *loadFeature{nullptr};
+        GstElement *elementObj{nullptr};
+
+        RIALTO_SERVER_LOG_DEBUG("Searching the properties of a class");
         {
-            for (auto i = propertiesToLookFor.begin(); i != propertiesToLookFor.end();)
+            // Try to obtain the class without instantiating an object...
+            GType elementType = m_gstWrapper->gstElementFactoryGetElementType(factory);
```

Comment:
This section of logic to handle the getting of the element type is duplicated quite a bit, maybe think about creating a separate function that would have that logic to avoid repetition and make the code cleaner

Reply:
Are you talking about these three lines that occur twice (once with XXX blank and once where XXX contains some additional code)...

                GType elementType = m_gstWrapper->gstElementFactoryGetElementType(factory);
                XXX
                if (elementType != G_TYPE_INVALID)
                    elementClass = m_glibWrapper->gTypeClassRef(elementType);
...you're thinking there should be a function for the last two lines?

Reply:
Yeh so some sort of helper function, I was thinking one to retrieve the element class, getElementClass, and another helper function to fetch properties from either the class or the object, getElementProperties maybe?